### ISSUE TITLE
target.json and linker script need a place to stay

### ISSUE THREAD

**Issue by mbr** [embed-rs/hello-embed-rs#1]
Regardless of how the `embed` crate will look like, we still need a convenient way to include linker scripts and target.json files. As a start, I could stuff the ones I have either into `embed-rs` or a separate repository.

Furthermore, I think they will be hard to share; initially the target.json one I wrote for the STM32F429i-Discovery was simply titled `thumbv7em-none-eabi.json`, which turned out to be not accurate enough. Now working on an STM32F103CBT6 project, the target file is the same except for the `-mcpu` setting (which changed from `-mcpu=cortex-m4` to `-mcpu=cortex-m3`).

It seems there is little one can do to prevent this fragmentation and we should be prepared to have one target per board/MCU.

Linker scripts are similarly an issue, if not worse: Due to changing Flash and SRAM-sizes depending on model, I can almost guarantee that there are changes necessary for each and every MCU.

To sum it up, each user project needs at least two extra files at the moment that we need to store and ship somewhere; I would personally attach them in a separate directory in the `embed-rs` repo; but if anyone argues for an extra repo (where they will most likely end up _eventually_), I won't object.
